src/state/dashboard/reducer.js

```diff
@@ -0,0 +1,196 @@
+// TODO: samples and experiments will most likely be moved into their own reducers in the future
+const initialState = {
+  stats: {},
+  samples: {},
+  samplesOverTime: [],
+  experiments: {},
+  experimentsOverTime: [],
+  jobs: {},
+  timeOptions: {
+    range: 'day',
+    timePoints: []
+  },
+  isLoading: false
+};
+
+const dashboardReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case 'DASHBOARD_REQUEST_SUCCESS': {
+      const { stats, samples, experiments } = action.data;
+      return {
+        ...state,
+        stats,
+        samples,
+        experiments
+      };
+    }
+    case 'DASHBOARD_TIME_REQUESTS_SUCCESS': {
+      const { samplesOverTime, experimentsOverTime, jobs } = action.data;
+      return {
+        ...state,
+        samplesOverTime,
+        experimentsOverTime,
+        jobs,
+        isLoading: false
+      };
+    }
+    case 'DASHBOARD_TIME_OPTIONS_SELECTED': {
+      return {
+        ...state,
+        isLoading: true
+      };
+    }
+    case 'DASHBOARD_TIME_OPTIONS_UPDATED': {
+      const { timeOptions } = action.data;
+      return {
+        ...state,
+        timeOptions
+      };
+    }
+    default: {
+      return state;
+    }
+  }
+};
+
+export default dashboardReducer;
+
+// chart selectors for creating chart data for individual charts on dashboard
+
+export function getTotalLengthOfQueuesByType(state) {
+  const stats = state.dashboard.stats;
+  return Object.keys(stats).map(jobType => {
+    return {
+      name: jobType.split('_')[0],
+      value: stats[jobType].open + stats[jobType].pending
+    };
+  });
+}
+
+export function getJobsByStatus(state) {
+  const stats = state.dashboard.stats;
+  return Object.keys(stats).reduce((accum, jobType) => {
+    accum[jobType] = Object.keys(stats[jobType]).reduce(
+      (allStatuses, status) => {
+        if (status !== 'total' && status !== 'average_time') {
+          allStatuses.push({
+            name: status,
+            value: stats[jobType][status]
+          });
+        }
+        return allStatuses;
+      },
+      []
+    );
+    return accum;
+  }, {});
+}
+
+function convertSecToMinHours(sec) {
+  const hours = Math.floor(sec / 3600),
+    minutes = Math.floor((sec % 3600) / 60);
```

Comment:
Since we only estimate the time to the minute level, I think it will be slightly better to use `Math.round()` here.